File: Data Structures & Algorithms/pow-x-n/submission-0.cs
public class Solution {
    public double MyPow(double x, int n) {
        if(n == 0) {
            return 1;
        }
        if(n < 0) {
            x = 1/x;
            n = -n;
        }
        double result = 1.0;
        double currentProduct = x;
        while(n > 0) {
            if(n%2 == 1) {
                result *= currentProduct;
            }
            currentProduct *= currentProduct;
            n/=2;
        }
        return result;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size(), product = 1;
        vector<int> result (n, 1);
        for(int i=1; i< n; i++) {
            int x = product * nums[i-1];
            result[i] = x;
            product = x; 
        }
        //[1,1,12,8]
        //[1,2,4,6]
        product = 1;
        for(int i = n-1; i>=0; i--) {
            int x = product * result[i];
            result[i] = x;
            product*= nums[i];
        }
        return result;
    }
};
